build_runner/lib/src/entrypoint/run_script.dart

```diff
@@ -170,6 +174,8 @@ class RunCommand extends BuildRunnerCommand {
       if (!exitCodeCompleter.isCompleted) {
         exitCodeCompleter.complete(ExitCode.success.code);
       }
+
+      await logSubscription.cancel();
```

Comment:
It looks like this should probably move to an outer try/finally - we have some other bailout conditions before this try block which could prevent this from running (which would make the vm hang on the subscription being open)